video/out/opengl/context_drm_egl.c

```diff
@@ -273,15 +285,25 @@ static void update_framebuffer_from_bo(struct ra_ctx *ctx, struct gbm_bo *bo)
     fb->fd     = p->kms->fd;
     fb->width  = gbm_bo_get_width(bo);
     fb->height = gbm_bo_get_height(bo);
-    uint32_t stride = gbm_bo_get_stride(bo);
-    uint32_t handle = gbm_bo_get_handle(bo).u32;
+    uint64_t modifier = gbm_bo_get_modifier(bo);
 
-    int ret = drmModeAddFB2(fb->fd, fb->width, fb->height,
+    uint32_t handles[4] = {0};
+    uint32_t strides[4] = {0};
+    uint32_t offsets[4] = {0};
+    uint64_t modifiers[4] = {0};
+
+    const int num_planes = gbm_bo_get_plane_count(bo);
+    for (int i = 0; i < num_planes; ++i) {
+        handles[i] = gbm_bo_get_handle_for_plane(bo, i).u32;
+        strides[i] = gbm_bo_get_stride_for_plane(bo, i);
+        offsets[i] = gbm_bo_get_offset(bo, i);
+        modifiers[i] = modifier;
+    }
+
+    int ret = drmModeAddFB2WithModifiers(fb->fd, fb->width, fb->height,
```

Comment:
This call will definitely need a fallback for the `modifier == DRM_FORMAT_MOD_INVALID` case.